### PR TITLE
initial commit: backend integrated

### PR DESCRIPTION
**Pull request by DanishOnGit**

### CODE REVIEW

File: router/products.router.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          success: false,
+          message: "Unable to add product!",
+          errMessage: err.message,
+        });
+    }
+  });
+
+router.param("productId", async (req, res, next, id) => {
+  try {
+    const product = await Product.findById({ _id: id });
```
Comment (MKumaran98) [inline router/products.router.js:42]: `product.findById(id)` will work too

File: index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+//404 handler: DO NOT move this. This should be last route!!
+
+app.use(errorHandler);
+app.use(routeNotFound)
+
+
+app.listen(3000, () => {
```
Comment (MKumaran98) [inline index.js:33]: port value can be accessed from env variables. this will help when moving to prod

File: router/wishlist.router.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          errMessage: err.message,
+        });
+    }
+  })
+  .post(async (req, res) => {
+    try {
+      const passedProduct = req.body;
+
+      const result = await WishlistItem.findOne({
+        _id: "60894cbb25ed850340db285f",
```
Comment (MKumaran98) [inline router/wishlist.router.js]: values shouldn't be hardcoded

**Review by MKumaran98 — COMMENTED** [review]
Hey danish I've suggested a few changes check it out

File: router/products.router.js (hunk 1 above)
Comment (DanishOnGit) [reply router/products.router.js:42]: okkk

File: index.js (hunk 2 above)
Comment (DanishOnGit) [reply index.js:33]: hmm..alright !

File: router/wishlist.router.js (hunk 3 above)
Comment (DanishOnGit) [reply router/wishlist.router.js]: yes i will try to change it

File: router/cart.router.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const router = express.Router();
+const { CartItem } = require("../models/cart.model");
+const { extend } = require("lodash");
+
+router
+  .route("/")
+  .get(async (req, res) => {
+    try {
+      const cartItems = await CartItem.findOne({}).populate(
```
Comment (DarkShadow20) [inline router/cart.router.js:10]: You can use find instead of findOne. Tell me if I am wrong?

File: router/products.router.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+          success: false,
+          message: "Could not fetch the desired product!",
+          errMessage: err.message,
+        });
+    }
+  })
+  .post(async (req, res) => {
+    try {
+      let { product } = req;
+      // const {productId} = req.params
```
Comment (DarkShadow20) [inline router/products.router.js]: You can remove this if not needed

File: router/products.router.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          message: "Could not update the desired product!",
+          errMessage: err.message,
+        });
+    }
+  })
+  .delete(async (req, res) => {
+    try {
+      const { product } = req;
+      await product.remove();
+      res.json({ success: true, message: "product removed", products });
```
Comment (DarkShadow20) [inline router/products.router.js:104]: during delete do we need to show product?

**Review by DarkShadow20 — CHANGES_REQUESTED** [review]

File: router/cart.router.js (hunk 4 above)
Comment (DanishOnGit) [reply router/cart.router.js:10]: yes both of them will work!

File: router/products.router.js (hunk 5 above)
Comment (DanishOnGit) [reply router/products.router.js]: yep!

File: router/products.router.js (hunk 6 above)
Comment (DanishOnGit) [reply router/products.router.js:104]: No its not necessary. We can exclude it if we want !

**Pull request merged**